### ISSUE TITLE
Error message shown after successful upgrade

### ISSUE THREAD

**Issue by simondate** [adaptlearning/adapt_authoring#2694]
## Expected Behaviour

I have been upgrading a few instances of the AAT and whenever I upgrade I get an error message which requires escape the command line (Ctrl + C), despite the install being fine.

Here is the error message:

```
Node dependencies installed successfully.
Web application built successfully.
Server has been updated successfully!
Adapt authoring tool upgraded to v0.11.1
(node:18620) DeprecationWarning: current Server Discovery and Monitoring engine is deprecated, and will be removed in a future version. To use the new Server Discover and Monitoring engine, pass option { useUnifiedTopology: true } to the MongoClient constructor.
MongoDB Connection Error: Error: Cannot find module '../mongo_client'
Require stack:
- /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js
- /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/mongo_client.js
- /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/index.js
- /home/simon/videotiiviste/node_modules/mongoose/lib/drivers/node-mongodb-native/binary.js
- /home/simon/videotiiviste/node_modules/mongoose/lib/drivers/node-mongodb-native/index.js
- /home/simon/videotiiviste/node_modules/mongoose/lib/index.js
- /home/simon/videotiiviste/node_modules/mongoose/index.js
- /home/simon/videotiiviste/node_modules/migrate-mongoose/src/lib.js
- /home/simon/videotiiviste/upgrade.js
(node:18620) UnhandledPromiseRejectionWarning: Error: Cannot find module '../mongo_client'
Require stack:
- /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js
- /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/mongo_client.js
- /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/index.js
- /home/simon/videotiiviste/node_modules/mongoose/lib/drivers/node-mongodb-native/binary.js
- /home/simon/videotiiviste/node_modules/mongoose/lib/drivers/node-mongodb-native/index.js
- /home/simon/videotiiviste/node_modules/mongoose/lib/index.js
- /home/simon/videotiiviste/node_modules/mongoose/index.js
- /home/simon/videotiiviste/node_modules/migrate-mongoose/src/lib.js
- /home/simon/videotiiviste/upgrade.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at loadClient (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js:24:14)
    at collectEvents (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js:237:21)
    at createServer (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js:430:27)
    at /home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js:349:14
    at parseConnectionString (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/core/uri_parser.js:682:3)
    at connect (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js:291:3)
    at ConnectOperation.execute (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/connect.js:202:5)
    at executeOperation (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/operations/execute_operation.js:83:26)
    at MongoClient.connect (/home/simon/videotiiviste/node_modules/mongoose/node_modules/mongodb/lib/mongo_client.js:219:10)
    at /home/simon/videotiiviste/node_modules/mongoose/lib/connection.js:708:12
    at new Promise (<anonymous>)
    at NativeConnection.Connection.openUri (/home/simon/videotiiviste/node_modules/mongoose/lib/connection.js:705:19)
    at Mongoose.createConnection (/home/simon/videotiiviste/node_modules/mongoose/lib/index.js:279:17)
    at new Migrator (/home/simon/videotiiviste/node_modules/migrate-mongoose/src/lib.js:48:46)
    at /home/simon/videotiiviste/lib/installHelpers.js:326:22
    at /home/simon/videotiiviste/node_modules/fs-extra/lib/mkdirs/mkdirs.js:37:14
    at FSReqCallback.oncomplete (fs.js:171:23)
(node:18620) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:18620) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I waited a few minutes and nothing happened I escaped out and the server seemed to be working fine.

I've tested multiple installations that errored like this and they all seemed to operate fine.

## Steps to Reproduce
1. Upgrade from AAT 0.10.5 to 0.11.1
1. Some FW I kept the same and others I upgraded too. Both led to the same error.

**Comment by danbuntu** [comment]
I found that doing 
npm install mongoclient

before running the upgrade would stop this error from appearing

**Comment by taylortom** [comment]
I *think* this message can be safely ignored, and is related to the switch from the old migration mechanism. We get the error because:
1. The upgrade script is initially run on an existing/old installation with the old migration script
2. During the upgrade, the new code is pulled down, and the npm dependencies reinstalled
3. **_Note that at this point, we're still running the old upgrade script, as it was executed prior to updating the code_**
4. When migrations are run towards the end of the upgrade, the script isn't able to find the npm modules used by the old migration process (as these have been removed and would have been cleaned out in 2.), and the migration fails

What we did to rectify this issue was to automatically run the new migrations when starting up the v0.11.* app for the first time.
